src/cdk/tree/tree-control.ts

```diff
@@ -0,0 +1,223 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {SelectionModel} from '@angular/cdk/collections';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {FlatNode, NestedNode} from './tree-data';
+import {CdkTreeNode} from './node';
+
+/**
+ * Tree control interface
+ */
+export interface TreeControl {
+  /** The expansion change event */
+  expandChange: BehaviorSubject<any>;
+
+  /** The expansion model */
+  expansionModel: SelectionModel<any>;
+
+  /** Whether the node is expanded or collapsed. Return true if it's expanded. */
+  expanded(node: any): boolean;
+
+  /** Get all decedents of a node */
+  getDecedents(node: any): any[];
+
+  /** Expand or collapse node */
+  toggle(node: any);
+
+  /** Expand one node */
+  expand(node: any);
+
+  /** Collapse one node */
+  collapse(node: any);
+
+  /** Expand all the nodes in the tree */
+  expandAll();
+
+  /** Collapse all the nodes in the tree */
+  collapseAll();
+
+  /** Toggle a node by expand/collapse it and all its decedents */
+  toggleDecedents(node: any);
+
+  /** Expand a node and all its decedents */
+  expandDecedents(node: any);
+
+  /** Collapse a ndoe and all its decedents */
+  collapseDecedents(node: any);
+}
+
+/** Base tree control. It has basic toggle/expand/collapse operations on a single node. */
+export abstract class BaseTreeControl<T extends FlatNode|NestedNode> implements TreeControl {
```

Comment:
What do you think of making a subdirectory for controls with separate files for `tree-control.ts`, `base-tree-control.ts`, `flat-tree-control.ts` and `nested-tree-control.ts`?